subprojects/logging/src/main/java/org/gradle/internal/logging/LoggingConfigurationBuildOptionFactory.java

```diff
@@ -161,7 +161,7 @@ public void applyTo(String value, LoggingConfiguration settings, Origin origin)
                 ConsoleOutput consoleOutput = ConsoleOutput.valueOf(consoleValue);
                 settings.setConsoleOutput(consoleOutput);
             } catch (IllegalArgumentException e) {
-                throw new CommandLineArgumentException(String.format("Unrecognized value '%s' for %s.", value, LONG_OPTION));
+                origin.handleInvalidValue(LONG_OPTION, value);
```

Comment:
:x: At the moment we only provide the Gradle property but once we support the command line option then this logic becomes incorrect. We'll need to handle different origins like you do in `ParallelismBuildOptionFactory.java`.